src/rssignores.cpp

```diff
@@ -90,17 +91,34 @@ void RssIgnores::dump_config(std::vector<std::string>& config_output) const
 
 bool RssIgnores::matches(RssItem* item)
 {
+	std::string prefix = "regex:";
+	int prefix_len = prefix.length();
 	for (const auto& ign : ignores) {
+		bool matched = false;
 		LOG(Level::DEBUG,
 			"RssIgnores::matches: ign.first = `%s' item->feedurl = `%s'",
 			ign.first,
 			item->feedurl());
-		if (ign.first == "*" || item->feedurl() == ign.first) {
-			if (ign.second->matches(item)) {
-				LOG(Level::DEBUG,
-					"RssIgnores::matches: found match");
-				return true;
+
+		if (!ign.first.compare(0, prefix_len, prefix)) {
+			std::string errorMessage;
+			std::string pattern = ign.first.substr(prefix_len, ign.first.length() - prefix_len);
+			auto regex = Regex::compile(pattern, REG_EXTENDED | REG_ICASE, errorMessage);
+			if (regex == nullptr) {
+				throw ConfigHandlerException(strprintf::fmt(
+						_("`%s' is not a valid regular expression: %s"),
+						pattern, errorMessage));
```

Comment:
This exception is not handled by codepaths that call `matches`. If the user specifies an invalid regex, Newsboat will just crash at runtime.

Let's move this check up into `handle_action()`. It already checks that the filter expression is correct; it can also try to compile the regex. Then in `matches` you can just rely on the fact that the regex was checked already.

(And test it like this: https://github.com/newsboat/newsboat/blob/4ddbcb50f122c846bb4e766ad03a6e6fcfe00a47/test/rssignores.cpp#L60-L63)